### DISCUSSION TITLE
Single app with conditional Route trees

### DISCUSSION THREAD

**Discussion by robertteevans** [TanStack/router#2936] [General]
I'm working on a single web application that needs to support both mobile and desktop experiences, but the requirements differ so significantly that I have two distinct sets of routes—one for mobile and one for desktop. I'd like to avoid making a separate app. Is there a recommended pattern for conditionally choosing between these two route trees at the root level based on the user's device type? I'd like to maintain all the great developer experience benefits—such as automatic route generation—provided by TanStack Router. Ideally, the CLI could intelligently detect the file being updated and know which route tree to update/generate (mobile vs desktop). Any guidance or best practices would be greatly appreciated.

**Comment by schiller-manuel** [comment]
can you please describe your use case in more detail?
do you for example have any overlap of the routes between mobile and desktop?
if not, what's the benefit of a single app?

**Reply by anthratush** [reply]
I think this one is a similar question: https://github.com/TanStack/router/discussions/3334

So I think the general question is: Is there a (preferred) way to conditionally/asynchronously include routes in the route tree?

The examples are:
1. Conditional route with a feature flag
2. A route based on the user's device type: `isMobile ? routeTreeMobile : routeTreeDesktop` (if I get the question right)
3. Protected and public routes: `isAuthenticated ? routeTreeProtected : routeTreePublic` like in this example with React Router: https://medium.com/@dennisivy/creating-protected-routes-with-react-router-v6-2c4bbaf7bc1c where we wrap protected routes with `PrivateRoutes` component
